.ci/ci_cache_components.sh

```diff
@@ -0,0 +1,215 @@
+		echo "Adding ${k}"
+		local real_path=$(readlink -f "${k}")
+		#Get version from binary (format: vmlinu{z|x}-4.19.24-25)
+		local kernel_binary=$(basename "${real_path}")
+		local current_kernel_version=$(echo "${kernel_binary}" | cut -d- -f2-)
+		build_cache "${kernel_binary}" "${current_kernel_version}"
+	done
+}
+
+# This builds the image by specifying the image version from
+# /usr/share/kata-containers and the image name
+cache_image_artifacts() {
+	local image_name="image"
+	local image_path="${kata_dir}/kata-containers.img"
+	local path=$(readlink -f "${image_path}")
+	local image_version=$(echo $(basename "${path}"))
+	build_cache "${image_name}" "${image_version}"
+	create_image_tar "${image_name}"
+}
+
+# This builds nemu by specifying the nemu version from the
+# runtime versions file and the nemu tar file name
+cache_nemu_artifacts() {
+	pushd "${tests_repo_dir}"
+	local nemu_version=$(get_version "assets.hypervisor.nemu.version")
+	local nemu_tar="kata-nemu-static.tar.gz"
+	popd
+	build_cache "${nemu_tar}" "${nemu_version}"
+}
+
+# This builds the initrd image by specifying the image initrd version
+# from /usr/share/kata-containers and the image initrd name
+cache_image_initrd_artifacts() {
+	local image_name="initrd"
+	local image_path="${kata_dir}/kata-containers-initrd.img"
+	local path=$(readlink -f "${image_path}")
+	local image_version=$(echo $(basename "${path}"))
+	build_cache "${image_name}" "${image_version}"
+	create_image_tar "${image_name}"
+}
+
+# This function receives as an argument if it is an image or an image initrd
+create_image_tar(){
+	local image_name="$1"
+	local getinfo=$(cat latest-"${image_name}")
+	tar -cJf "${getinfo}.tar.xz" "${getinfo}"
+	sha256sum "${getinfo}.tar.xz" > "sha256sum-${image_name}-tarball"
+	sha256sum -c "sha256sum-${image_name}-tarball"
+	rm "${getinfo}"
+}
+
+# This function receives as arguments the component name (qemu, nemu, kernel, etc)
+# as well as the version
+build_cache() {
+	local component_name="$1"
+	local component_version="$2"
+	local check_image=$(echo "${component_version}" | grep kata || true)
+	local check_nemu=$(echo "${component_name}" | grep nemu || true)
+	local check_qemu=$(echo "${component_name}" | grep qemu || true)
+	local check_kernel=$(echo "${component_name}" | grep vmlinu || true)
+	local check_initrd=$(echo "${component_name}" | grep initrd || true)
+
+	# Verify which kind of image we are using (initrd or not)
+	if [ ! -z "${check_image}" ]; then
+		if [ ! -z "${check_initrd}" ]; then
+			image_path="${kata_dir}/kata-containers-initrd.img"
+			image_name="initrd"
+		else
+			image_path="${kata_dir}/kata-containers.img"
+			image_name="image"
+		fi
+		path=$(readlink -f "${image_path}")
+		echo $(basename "${path}") > "latest-${image_name}"
+		sudo cp "${path}" "${kata_dir}/osbuilder-${image_name}.yaml"  .
+	else
+		echo "${component_version}" >  "latest"
+	fi
+
+	# In the case of nemu and qemu we have the tar file in
+	# a specific location /tmp
```

Comment:
@jodh-intel that is the explanation

Reply:
Can you drop the mention of `/tmp` in the comment as it's now clear from the value of `KATA_TESTS_CACHEDIR`.